### ISSUE TITLE
extend check for image types with svg and webp

### ISSUE THREAD

**Issue by Klap-in** [dokuwiki/dokuwiki#4012]
More like https://github.com/dokuwiki/dokuwiki/commit/29d63aa571c7ffaccb323c0bb2a6a230f95c0d5d

e.g. 
https://codesearch.dokuwiki.org/xref/dokuwiki/lib/scripts/media.js#716
https://codesearch.dokuwiki.org/xref/dokuwiki/inc/search.php#221
https://codesearch.dokuwiki.org/xref/dokuwiki/inc/media.php#1129 and other
https://codesearch.dokuwiki.org/xref/dokuwiki/inc/Ui/MediaDiff.php#90
https://codesearch.dokuwiki.org/xref/dokuwiki/inc/parser/xhtml.php#1814
Probably more…

Does JpegMeta has limited support for file types? I guess so, see for replacement suggestions in #1970

**Comment by Klap-in** [comment]
It is more then just checking extension probably. 
My impression is that these checks here are for switching between handling as image or as non-image. For that check indeed a function might be helpful as well. 
So far I remember it right now, checking if a file or image is allowed, or should be only downloadable, is already happening at another place, and is indeed based at the mime config.

If generalizing the check for if file should be treated as image, then all the occurrences should support all these files. I guess at most places this is the case, but maybe e.g. jpegmeta could be a limitation.
So this update needs a bit of testing.

**Comment by Klap-in** [comment]
If simple check, without involving mime config, is working, I propose not to add it. Only add it if really needed, please keep things simple as possible.

**Comment by saggi-dw** [comment]
> If simple check, without involving mime config, is working, I propose not to add it. Only add it if really needed, please keep things simple as possible.

Yes, as simple as possible. I was thinking more of a central place for allowed formats. That new formats would not have to be specified extra. There are many `preg_match` with allowed extensions.
After the idea e.g.: `is_valid_pictureext (returns true/false)` that the function gets the data from the `mime.conf` was only an idea, since there the supported formats are already listed. 

I was thinking that in the future you would then only have to change the supported formats in one place.

Optionally, you could also build or use tests that check images for consistency (like media_contentcheck in media.php). But that would no longer be as simple as possible.
